analyzers/tests/SonarAnalyzer.UnitTest/MetadataReferences/NuGetMetadataFactory.Package.cs

```diff
@@ -186,8 +166,11 @@ private string NextCheckFilePath()
                 // The file containing the next-check timestamp is stored in folder of the latest version of the package.
                 const string NextUpdateFileName = "NextCheckForUpdate.txt";
                 var directory = SortedPackageFolders().LastOrDefault();
-                return directory == null ? null : Path.Combine(directory, NextUpdateFileName);
+                return directory == null ? null : Path.Combine(directory, "..", NextUpdateFileName);
```

Comment:
That should be same as this (the comment is outdated anyway)
```
private string NextCheckFilePath() =>
  Path.Combine(PackagesFolder, id, "NextCheckForUpdate.txt");
```

Reply:
Why does it need the `SortedPackageFolders`? I think we can really delete most of it and directly generate the returned value.

Reply:
This call is useless indeed. But we still need to verify that package folder exists to not write file to non-existing directory.

Reply:
Okay, this is very corner case, indeed, I will drop it, otherwise we could hide the package unpacking failure